packages/collectxyz/src/nft.rs

```diff
@@ -195,6 +195,27 @@ impl XyzTokenInfo {
             },
         }
     }
+
+    pub fn base64_token_image(&self) -> String {
+        base64_token_image(&self.extension.coordinates)
+    }
+}
+
+pub fn base64_token_image(coords: &Coordinates) -> String {
+    let svg = format!(
+        r#"<svg xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="xMinYMin meet" viewBox="0 0 240 240"><g class="container"><rect style="width:240px;height:240px;fill:#000;"/><text x="120" y="120" dominant-baseline="middle" text-anchor="middle" style="fill:#fff;font-family:serif;font-size:16px;text-align:center;">[{}, {}, {}]</text></g></svg>"#,
+        coords.x, coords.y, coords.z
+    );
+    let base64_uri = format!("data:image/svg+xml;base64,{}", base64::encode(svg));
+    base64_uri
+}
+
+pub fn full_token_id(numeric_token_id: String) -> String {
+    format!("xyz #{}", numeric_token_id)
+}
+
+pub fn numeric_token_id(full_token_id: String) -> String {
+    full_token_id[5..].to_string()
 }
```

Comment:
should we have some regex/assertions here to make sure the input and out put are what we expect at runtime. I.e. I pass already full_token_id to full_token_id function, do we want to to recognize that and not break? Ideally we can fail/throw an error to avoid potentially saving incorrect output

Reply:
yeah, good idea!